code/modules/mob/living/simple_animal/hostile/human/nazi.dm

```diff
@@ -0,0 +1,372 @@
+	max_n2 = 0
+	minbodytemp = 0
+	maxbodytemp = 1000
+
+
+	retreat_distance = 4
+	minimum_distance = 3
+
+	light_range = 4
+	light_color = "#FF2222"
+
+	ammo = 6
+	reloads = 2
+	ranged_cooldown_cap = 2
+
+	var/datum/effect/effect/system/trail/ion_trail
+
+/mob/living/simple_animal/hostile/humanoid/nazi/spacetrooper/New()
+	. = ..()
+	ion_trail = new /datum/effect/effect/system/trail()
+	ion_trail.set_up(src)
+	ion_trail.start()
+
+/mob/living/simple_animal/hostile/humanoid/nazi/spacetrooper/Process_Spacemove(var/check_drift = 0)
+	return 1
+
+/mob/living/simple_animal/hostile/humanoid/nazi/spacetrooper/afterShoot()
+	ammo--
+	new /obj/item/ammo_casing/a357(loc,1)
+	if(ammo <= 0)
+		if(reloads > 0)
+			spawn(10)
+				playsound(src, 'sound/weapons/revolver_spin.ogg', 100, 1)
+			reloads--
+			new /obj/item/ammo_storage/speedloader/a357/empty(loc)
+			ammo = 6
+		else
+			//we be going berserker
+			emote("me",1,"puts away his mateba and tightens the grip on his knife.")
+			melee_damage_lower = 40
+			melee_damage_upper = 40
+			retreat_distance = 0
+			minimum_distance = 0
+			ranged = 0
+			spawn(10)
+				if(health > 0)
+					say("Come over here. I promise I will kill you!")
+
+
+///////////////////////////////////////////////////////////////////HOLYSHIT/////////////
+
+/mob/living/simple_animal/hostile/mechahitler
+	name = "Mecha Hitler"
+	desc = "Heil Myself!"
+	icon = 'icons/mob/hostile_humanoid.dmi'
+	icon_state = "mechahitler"
+	icon_living = "mechahitler"
+	icon_dead = "mechahitler"
+	icon_gib = "syndicate_gib"
+
+	move_to_delay = 20
+
+	speak = list("Die, Allied schweinhund!","Die Welt ist unser!","Guten Tag!" ,"Scheise!")
+	speak_emote = list("says")
+	emote_hear = list("says")
+	emote_see = list("hums")
+	attacktext = "crushes"
+	attack_sound = 'sound/weapons/heavysmash.ogg'
+	speak_chance = 20
+
+	health = 1000
+	maxHealth = 1000
+
+	melee_damage_lower = 40
+	melee_damage_upper = 80
+
+	faction = "nazi"
+
+	ranged = 1
+	rapid = 1
+	retreat_distance = 0
+	minimum_distance = 2
+	projectiletype = /obj/item/projectile/bullet/gatling
+	projectilesound = 'sound/weapons/gatling_fire.ogg'
+	casingtype = null
+	ranged_cooldown_cap = 1
+
+	environment_smash = 3
+
+	min_oxy = 0
+	max_oxy = 0
+	min_tox = 0
+	max_tox = 0
+	min_co2 = 0
+	max_co2 = 0
+	min_n2 = 0
+	max_n2 = 0
+	minbodytemp = 0
+	maxbodytemp = 5000
+
+	light_range = 2
+	light_color = "#88BB88"
+
+	var/datum/effect/effect/system/trail/ion_trail
+
+/mob/living/simple_animal/hostile/mechahitler/Shoot(var/atom/target, var/atom/start, var/mob/user, var/bullet = 0)
+	if(..())
+		new /obj/item/ammo_casing_gatling(loc)
+
+/mob/living/simple_animal/hostile/mechahitler/New()
+	. = ..()
+	ion_trail = new /datum/effect/effect/system/trail()
+	ion_trail.set_up(src)
+	ion_trail.start()
+
+/mob/living/simple_animal/hostile/mechahitler/Process_Spacemove(var/check_drift = 0)
+	return 1
+
+/mob/living/simple_animal/hostile/mechahitler/Die()
+	dir = 2
+	say("Eva, auf wiedersehen!")
+	sleep(10)
```

Comment:
I feel you're gonna need a set waitfor = 1 at the start of this proc if you're gonna use a sleep, lest we end up with sleepy zombie syndrome again